### PR TITLE
added CinemaHallController, MovieController,

### PR DESCRIPTION
**Pull request by Ilya-brinchuk**
MovieSessionController

### CODE REVIEW

**Review by engineer027 — APPROVED** [review]

File: src/main/java/com/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private final MapperDto<CinemaHallResponseDto, CinemaHall, CinemaHallRequestDto> mapperDto;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/CinemaHallController.java]:
```suggestion
    private final DtoMapper<CinemaHallResponseDto, CinemaHall, CinemaHallRequestDto> dtoMapper;
```

File: src/main/java/com/cinema/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                .map(mapperDto::replaceWithDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    public void create(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(mapperDto.replaceWithEntity(movieSessionRequestDto));
+    }
+
+    @PutMapping("{id}")
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/MovieSessionController.java]:
```suggestion
    @PutMapping("/{id}")
```
is it working the same or no?

File: src/main/java/com/cinema/controller/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    public void updateMovieSession(@PathVariable Long id,
+                                   @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = mapperDto.replaceWithEntity(movieSessionRequestDto);
+        movieSession.setId(id);
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping("{id}")
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/MovieSessionController.java]:
```suggestion
    @DeleteMapping("/{id}")
```
here as well

File: src/main/java/com/cinema/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    public void deleteMovieSession(@PathVariable Long id,
+                                   @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controller/MovieSessionController.java]: let's not pass the body here - only id

File: src/main/java/com/cinema/dao/MovieSessionDao.java [hunk 5]
```diff
@@ -8,4 +8,8 @@
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
     MovieSession add(MovieSession session);
+
+    MovieSession update(MovieSession movieSession);
+
+    MovieSession delete(MovieSession movieSession);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/MovieSessionDao.java]:
```suggestion
    void delete(Long movieSessionId);
```

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 6]
```diff
@@ -43,6 +43,52 @@ public MovieSession add(MovieSession movieSession) {
...
+    }
+
+    @Override
+    public MovieSession delete(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(movieSession);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java:78]: let's use `load()` method to get the session by id before deleting it

File: src/main/java/com/cinema/model/dto/CinemaHallResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package com.cinema.model.dto;
+
+public class CinemaHallResponseDto {
+    private Long id;
+    private String description;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/CinemaHallResponseDto.java:5]: let's return capacity as well

File: src/main/java/com/cinema/model/dto/MovieResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package com.cinema.model.dto;
+
+public class MovieResponseDto {
+    private Long id;
+    private String title;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/MovieResponseDto.java:5]: here description as well

File: src/main/java/com/cinema/service/MapperDto.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package com.cinema.service;
+
+public interface MapperDto<T,E, Q> {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/MapperDto.java]:
```suggestion
public interface DtoMapper<T, E, Q> {
```

File: src/main/java/com/cinema/service/MapperDto.java [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+    T replaceWithDto(E entity);
+
+    E replaceWithEntity(Q requestDto);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/MapperDto.java]:
```suggestion
    T mapToDto(E entity);

    E mapToEntity(Q requestDto);
```

File: src/main/java/com/cinema/service/mapper/CinemaHallMapper.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    @Override
+    public CinemaHall replaceWithEntity(CinemaHallRequestDto requestDto) {
+        CinemaHall cinemaHall = new CinemaHall();
+        cinemaHall.setDescription(requestDto.getDescription());
+        cinemaHall.setCapacity(requestDto.getCapacity());
+        return cinemaHall;
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/CinemaHallMapper.java]: redundant empty line

File: src/main/java/com/cinema/service/mapper/MovieMapper.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    public Movie replaceWithEntity(MovieRequestDto requestDto) {
+        Movie movie = new Movie();
+        movie.setTitle(requestDto.getTitle());
+        movie.setDescription(requestDto.getDescription());
+        return movie;
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/MovieMapper.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/service/impl/MovieSessionServiceImpl.java [hunk 13]
```diff
@@ -22,6 +23,22 @@ public MovieSession add(MovieSession session) {
...
     }
 
+    @Override
+    public MovieSession update(MovieSession movieSession) {
+        return movieSessionDao.update(movieSession);
+    }
+
+    @Override
+    public void delete(Long id) {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/impl/MovieSessionServiceImpl.java]: redundant empty line

File: src/main/java/com/cinema/service/impl/MovieSessionServiceImpl.java [hunk 14]
```diff
@@ -22,6 +23,22 @@ public MovieSession add(MovieSession session) {
...
+        movieSessionDao.delete(id);
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id).orElseThrow(() -> new DataProcessingException(""));
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/impl/MovieSessionServiceImpl.java]: let's add informative message here?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**